core/trino-main/src/main/java/io/trino/operator/scalar/StringFunctions.java

```diff
@@ -406,6 +406,10 @@ public static Block split(@SqlType("varchar(x)") Slice string, @SqlType("varchar
             if (splitIndex < 0) {
                 break;
             }
+            if (delimiter.length() == 0) {
```

Comment:
I think we need to remove `checkCondition(delimiter.length() > 0` as well.
I hope the test is catching it